### DISCUSSION TITLE
[WASM] How to add a assembly to wasm to make it availble to load at runtime

### DISCUSSION THREAD

**Discussion by SongOfYouth** [unoplatform/uno#12855] [Q&A] [answered]
we allways need to reflect a assembly at runtime when use IOC, it's nothing to say on windows, but how to approch to it on wasm?

**Comment by jeromelaban** [comment]
Reflection is working on WebAssembly the same way it does on other platforms. What's not working for you there?

**Comment by SongOfYouth** [comment]
I want to reflect it when the app is starting, for windows, we usually place a assembly in a sub folder, but where is the root for wasm and how to ensure the dll can be scan when the web load?



---Original---
From: "Jérôme ***@***.***&gt;
Date: Tue, Jul 11, 2023 20:14 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [unoplatform/uno] [WASM] How to add a assembly to wasm to make itavailble to load at runtime (Discussion #12855)




 
Reflection is working on WebAssembly the same way it does on other platforms. What's not working for you there?

**Reply by jeromelaban** [reply]
You can place assemblies as `Content` files, fetch them using [`StorageFile.GetFileFromApplicationUriAsync`](https://platform.uno/docs/articles/features/windows-storage.html#support-for-storagefilegetfilefromapplicationuriasync), then load it using `Assembly.LoadFrom`. You will not be able to scan a directory to read the files, as they're not available as standard files.

**Reply by SongOfYouth** [reply]
thanks for reply. in my case, i dot know what assembly will be load when building my app since it 's just a  frame and will load assemblies base on config file. Is there a way to do the same as setting the assemblies as `Content` files after building?

**Reply by jeromelaban** [reply, marked as answer]
If you need to load assemblies using a config file, you'll need to do a round of [StorageFile.GetFileFromApplicationUriAsync](https://platform.uno/docs/articles/features/windows-storage.html#support-for-storagefilegetfilefromapplicationuriasync) to download the files and write them to the browser's in-memory file system, then point your assemly loading logic to the folder you created when downloading the files.
